test/app_profiler/storage/google_cloud_storage_test.rb

```diff
@@ -22,6 +22,31 @@ def teardown
         end
       end
 
+      [false, true].each do |forward_metadata_on_upload|
+        test "sets metadata when forward_metadata_on_upload = #{forward_metadata_on_upload}" do
+          old_forward_metadata_on_upload = AppProfiler.forward_metadata_on_upload
+          AppProfiler.forward_metadata_on_upload = forward_metadata_on_upload
+
+          file = Google::Cloud::Storage::File.new
+          profile = profile_from_stackprof
+          bucket = Google::Cloud::Storage::Bucket.new
+
+          expected_metadata = nil
+          if forward_metadata_on_upload
+            expected_metadata = profile.metadata
+          end
```

Comment:
This is a bit of a code smell, you're duplicating the logic in your test to save a few lines, and that can make the test brittle over time. I'd consider changing `with_mock_gcs_bucket` to assert properties you need, and introducing a method like `with_stubbed_process_queue_thread` below for your new config to toggle global values.